projects/igniteui-angular/src/lib/data-operations/data-util.ts

```diff
@@ -30,24 +30,31 @@ export enum DataType {
  * @hidden
  */
 export class DataUtil {
-    public static sort<T>(data: T[], expressions: ISortingExpression[], sorting: IgxSorting = new IgxSorting()): T[] {
+    public static sort<T>(data: T[], expressions: ISortingExpression[], sorting?: IGridSortingStrategy): T[] {
+        if (!sorting) {
+            sorting = new IgxSorting();
+        }
         return sorting.sort(data, expressions);
     }
 
     public static treeGridSort(hierarchicalData: ITreeGridRecord[],
         expressions: ISortingExpression[],
-        parent?: ITreeGridRecord): ITreeGridRecord[] {
+        parent?: ITreeGridRecord,
+        sorting?: IGridSortingStrategy): ITreeGridRecord[] {
         let res: ITreeGridRecord[] = [];
         hierarchicalData.forEach((hr: ITreeGridRecord) => {
             const rec: ITreeGridRecord = DataUtil.cloneTreeGridRecord(hr);
             rec.parent = parent;
             if (rec.children) {
-                rec.children = DataUtil.treeGridSort(rec.children, expressions, rec);
+                rec.children = DataUtil.treeGridSort(rec.children, expressions, rec, sorting);
             }
             res.push(rec);
         });
 
-        res = DataUtil.sort(res, expressions, new IgxDataRecordSorting());
+        if (!sorting) {
+            sorting = new IgxDataRecordSorting();
+        }
```

Comment:
Again provide a default value in the parameter definition:  `sorting = new IgxDataRecordSorting()`. This way the parameter doesn't have to be optional.

Reply:
Same comment as for the IgxSorting on top.